source/base/timer.cc

```diff
@@ -37,6 +38,83 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  namespace Timer
+  {
+    namespace
+    {
+      /**
+       * Type trait for checking whether or not something is a duration.
+       */
+      template <typename T>
+      struct is_duration : std::false_type {};
```

Comment:
None; this is just a check on my own work to make sure I never pass in a non-duration.